services/dao-service.ts

```diff
@@ -0,0 +1,155 @@
+import { 
+  Network_v2,
+  OraclesResume,
+  Web3Connection
+} from "@taikai/dappkit";
+import getConfig from "next/config";
+
+import { NetworkParameters } from "types/dappkit";
+
+const { publicRuntimeConfig } = getConfig();
+
+export default class DAO {
+  private _web3Connection: Web3Connection;
+  private _network: Network_v2;
+
+  get web3Connection() { return this._web3Connection; }
+  get network() { return this._network; }
+
+  constructor() {
+    if (!publicRuntimeConfig?.web3ProviderConnection) 
+      throw new Error("Missing web3ProviderConnection in next.config.js");
+
+    this._web3Connection = new Web3Connection({
+      web3Host: publicRuntimeConfig.web3ProviderConnection
+    });
+  }
+
+  async loadNetwork(networkAddress: string = publicRuntimeConfig?.contract?.address, 
+                    shouldReturn?: boolean): Promise<Network_v2 | boolean> {
+    try {
+      if (!networkAddress) throw new Error("Missing Network Contract Address");
+
+      const network = new Network_v2(this.web3Connection, networkAddress);
+
+      await network.loadContract();
+
+      if (!shouldReturn) this._network = network;
+
+      return network;
+    } catch (error) {
+      console.log("Error loading network: ", error);
+    }
+
+    return false;
+  }
+
+  async start(): Promise<boolean> {
+    try {
+      await this.web3Connection.start();
+
+      return true;
+    } catch (error) {
+      console.log("Error starting: ", error);
+    }
+
+    return false;
+  }
+
+  async connect(): Promise<boolean> {
+    try {
+      await this.web3Connection.connect();
+
+      await this.loadNetwork(this.network?.contractAddress);
+
+      return true;
+    } catch (error) {
+      console.log("Error logging in: ", error);
+    }
+
+    return false;
+  }
+
+  async getAddress(): Promise<string> {
+    return this.web3Connection.getAddress();
+  }
+
+  async getChainId(): Promise<number> {
+    return this.web3Connection.web3.eth.getChainId();
+  }
+
+  async getNetwork(networkAddress: string = undefined): Promise<Network_v2> {
```

Comment:
This function can't be called this many times; If we're rewriting this, we need to make sure that when we `new` this class, we have to make sure that a network is already loaded.